### PR TITLE
Unambiguous limitation about ASCII

### PR DESCRIPTION
**Pull request by oliviercailloux**
Issue: #8551.

Commit message should be: “Issue #8551: Consider relaxing the stated limitation about UTF-8” (sorry, just saw it when opening the PR).

The issue does not have the “approved” label, but I understood the comment there (asking to suggest better wording and referring to the source code of the website) as inviting me to introduce a PR. I hope this is okay.

### CODE REVIEW

**Comment by romani** [comment]
@oliviercailloux , please make commit message follow our rules pleas make it `doc: .....`

**Comment by romani** [comment]
rebased, mesage is corrected.

**Comment by pbludov** [comment]
Build failed:
https://ci.appveyor.com/project/Checkstyle/checkstyle/builds/36136239/job/vk2j4th52ahr2s8b#L780

@oliviercailloux please update the commit message. It should be just one single line.

**Comment by oliviercailloux** [comment]
I am sorry but frankly, it is a bit too much to ask to your contributors, for a single line change. I really tried to follow instructions, but something was unclear (don’t recall what) and so it failed. That’s enough. Someone from the team, who knows already how to comply to these instructions, can do this change in, literally, 30 seconds. For me, it’s much more, because I need to learn the rules first.

Again, I understand perfectly that there are rules to collaborate code, but for a single line change in a string, this is utterly absurd. Everybody (including the reviewers) is losing time.

Please close this and leave the phrasing as it is if nobody is interested in a better phrasing. It’s a small detail, after all.

**Comment by romani** [comment]
commit is updated.

**Comment by romani** [comment]
@oliviercailloux ,

>  I really tried to follow instructions, but something was unclear (don’t recall what) and so it failed. 

it would be awesome to get feedback from you on this.
We try to reduce threshold of entrance to project to contribute, we are already blind, feedback from a side is much appreciated.

**Comment by oliviercailloux** [comment]
> it would be awesome to get feedback from you on this.

I just tried to create a new PR and now remember better what my concerns were. 

Here are the rules that appear _after having asked GitHub for a new PR_.

```
Rules:
1) The issue you are trying to fix/resolve has to have the "approved" label.
2) Put in the description of Pull Request the reference to an issue if it exists.
   Example: "Issue: #XXXXXX"
3) Commit message should adhere to the following rules:
   a) MUST match one of the following patterns:
      ^Issue #\\d+: .*$
      ^Pull #\\d+: .*$
      ^(minor|config|infra|doc|spelling|dependency): .*$
   b) MUST contain only one line of text
   c) MUST NOT end with a period, space, or tab
   d) MUST be less than or equal to 200 characters

To avoid multiple iterations of fixes and CIs failures, please read
https://checkstyle.org/contributing.html

ATTENTION: We are not merging Pull Requests that are not passing our CIs,
but we will help to resolve issues.

Thanks for reading, remove whole this message and type what you need.
```

I had a problem with 1). In my case someone from the checkstyle team had explicitly asked during the discussion in the related issue that I create a PR, but there was no “approved” label. I supposed that this instruction therefore was not applicable to me, but still, this creates an uncomfortable feeling of insecurity, because I feared that someone (or some robot) might blindly follow rules and reject my PR just because of the missing label. This reduced my willingness to spend time and energy in precisely following the rest of the rules, considering that the whole PR was perhaps doomed to fail because of this first point.

The rule 2) is fine: it is easy to understand its purpose and easy to follow.

The rule 3) is especially problematic. I had created the commit before introducing the PR (of course). So the commit message was in my history already. Complying with this rule is therefore not easy: it requires to cancel the PR, go back and amend the commit, open a PR again. Furthermore, I have no way, in my case (concerning a change of documentation following a discussion in a given Issue), to know whether I must use the first or the third pattern in 3)a). Thus perhaps I should cancel my PR and first discuss those points with the checkstyle team before introducing it? (Who’s saving time thanks to these rules then?) Also, contrary to 2), it is hard to find this requirement reasonable. Note that my commit message did reference the issue (but not matching the required pattern), I did this naturally. In my opinion, the role of an external occasional contributor is to make their proposal clear and make reasonable efforts towards this (such as in following rule 2)), preferably rules that regular open source contributors will know, not rules overly specific to a given project. The precise bookkeeping of the relations of commits to issues by following regular expressions or any other very specific means belongs to the team.

Only my 2 cents.

**Review by pbludov — APPROVED** [review]

**Comment by romani** [comment]
@oliviercailloux , thank a lot for your feedback, I really appreciate your time to give us feedback.

@rnveach , please review this easy PR and approve it. I will merge this PR myself after I thoroughly review feedback and make few action items to improve. You are also welcome with suggestions.

--------------

I will put here my thoughts ..... a bit later ...

**Review by romani — APPROVED** [review]

**Comment by romani** [comment]
@oliviercailloux ,

> 1) The issue you are trying to fix/resolve has to have the "approved" label.

What if we change wording like:
`1) If your fix addressing existing issue please make sure it has "approved" label, if it does not such label you can continue to create PR but will NOT be accepted until issue get "approved" label.`

We can probably remove commit message rules, too much text to read, prone to be ignored, as it is now enforced by CI validation, so it unavoidable to follow it.
But as PR going to me created, it means commit message is already done and followed rules, or created without execution of build (`mvn test`) on local so it is still too late and It will unlikely to stop contributor to finish PR creation as he always can update/polish PR after creation. CI violations will guide contributor on what to do. If contributor is not experienced in CIs, he will wait for maintainers (as usually happens).

@rnveach , do you agree to remove commit message block ? We still see very high rate of PRs where contributors do not read anything and only CI failure can guide some of them.

File: src/xdocs/writingchecks.xml
Comment (rnveach) [inline src/xdocs/writingchecks.xml:666]: Why the change from `code` to `identifiers`?

An identifier is things the user name like classes and variables.

I assume "code" expands to keywords. It is possible to write these keywords with unicode which we don't support.

The following example compiles on Eclipse:
````
public cl\u0061ss Test {}
````
but can't pass Checkstyle.

I don't see this change being correct.

**Review by rnveach — CHANGES_REQUESTED** [review]

**Comment by rnveach** [comment]
> do you agree to remove commit message block ?

Which block? These?
https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/internal/CommitValidationTest.java#L280-L287

I don't think we should remove any of them. User can find out failures like this by running test suite before submitting PR and make corrections. If user didn't run the test suite, then nothing we can do about it as we have many other checks besides commit validation which can be flagged in what some may consider unusual places. If they ran suite without creating commit, then we can add extra check that there are no uncommitted changes. Not all our checks require server CI. The only server only CIs are the most intensive ones.

**Comment by romani** [comment]
we need to use JDK specification 

`Java tokens(Identifier, Keyword) should be written with ASCII characters only, no Unicode escape support in keywords and no Unicode support in identifiers.`

with embeded links:
https://docs.oracle.com/javase/specs/jls/se8/html/jls-3.html#jls-3.5
https://docs.oracle.com/javase/specs/jls/se8/html/jls-3.html#jls-3.3
https://docs.oracle.com/javase/specs/jls/se8/html/jls-3.html#jls-3.1

**Comment by romani** [comment]
@rnveach 

> Which block? 

from https://github.com/checkstyle/checkstyle/blob/master/.github/PULL_REQUEST_TEMPLATE.md

**Comment by rnveach** [comment]
> from https://github.com/checkstyle/checkstyle/blob/master/.github/PULL_REQUEST_TEMPLATE.md

Again, which block? All of it? Are we removing MD but leaving commit message test?

File: src/xdocs/writingchecks.xml
Comment (rnveach) [reply src/xdocs/writingchecks.xml:666]: See @romani 's message at https://github.com/checkstyle/checkstyle/pull/8565#issuecomment-723456747

**Comment by romani** [comment]
@rnveach , sorrry for being unclear, can not send link to exact line in MD (limitation of github) 

I suggest to remove this block:
```
Commit message should adhere to the following rules: a) MUST match one of the following patterns: ^Issue #\d+: .$ ^Pull #\d+: .$ ^(minor|config|infra|doc|spelling|dependency): .*$ b) MUST contain only one line of text c) MUST NOT end with a period, space, or tab d) MUST be less than or equal to 200 characters
```

**Comment by rnveach** [comment]
> I suggest to remove this block:

If we are removing that block without touching the test, then it doesn't make much sense IMO as I see that more as a warning the test will come and ensure validation. If you really want to remove text but leave test as is, then ok.
If we are removing the test too, then see https://github.com/checkstyle/checkstyle/pull/8565#issuecomment-723455832 .

File: src/xdocs/writingchecks.xml
Comment (romani) [reply src/xdocs/writingchecks.xml:666]: fixed.

**Comment by romani** [comment]
I suggest to remove this text block only, and keep UT do validation. 
I noticed that even we have this block, not that much people follow it and create PR no matter what, with further updates if they aware of what is CI  and can read logs of CI failure.
I might be wrong but if they see this block of text as not done in their commit, it does not stop them, so it is ignored most of the time. Nobody can ignore CI failure :).

**Comment by romani** [comment]
@rnveach , we will continue PR template discussion at https://github.com/checkstyle/checkstyle/issues/8969

lets  finish with this doc update. Please finalize review.

**Comment by romani** [comment]
Github, regenerate wesite

**Review by rnveach — APPROVED** [review]
Formatting is odd, but ok.

**Comment by romani** [comment]
GitHub, rebase

**Pull request merged**
